Update color of continue button for new learner view #960

edallison opened the pull request:
I wanted to leave the background-color styling out of the html however I believe it is getting overridden somewhere.

seanlip commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -53,8 +53,9 @@ md-input-group.md-default-theme label
   margin-bottom: 0;
 }
 
-.md-button-success {
```
Don't overwrite this; we use it elsewhere. The command I used to verify that was:
```
    grep "md-button-success" . -r --exclude-dir=third_party
```

edallison replied:
Done

seanlip commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -53,8 +53,9 @@ md-input-group.md-default-theme label
-.md-button-success {
-  color: #009688;
+.oppia-learner-continue-button {
+  color: #FFFFFF;
```
I think we're now generally using lowercase: #fff

edallison replied:
Done

seanlip commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -53,8 +53,9 @@ md-input-group.md-default-theme label
 
-.md-button-success {
-  color: #009688;
+.oppia-learner-continue-button {
```
Use `.md-button.oppia-learner-continue-button` instead. The reason you're getting a different colour than expected is due to the CSS specificity rules, which you can read about [here](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity). The extra class fixes this.

By the way, you can see what CSS styles are being applied to an element using the browser console: https://developer.chrome.com/devtools/docs/elements-styles . This allows for easy in-browser experimentation.

edallison replied:
Done

seanlip commented on extensions/interactions/Continue/Continue.html:
```diff
@@ -1,5 +1,5 @@
 <script type="text/ng-template" id="interaction/Continue">
-  <md-button class="md-button-success" ng-click="submitAnswer()"><[buttonText]></md-button>
+  <md-button class="oppia-learner-continue-button" style="background-color: rgb(29, 59,164)" ng-click="submitAnswer()"><[buttonText]></md-button>
```
Shouldn't need inline style here.

edallison replied:
Done

seanlip commented on extensions/skins/conversation_v1/player.html:
```diff
@@ -99,7 +99,7 @@ <h4 class="conversation-skin-responses-dropdown-container" ng-click="toggleShowP
               <img class="conversation-skin-user-avatar img-circle" ng-src="<[profilePicture]>">
               <div ng-if="waitingForContinueButtonClick">
-                <md-button class="md-button-success protractor-test-continue-to-next-card-button" focus-on="<[CONTINUE_BUTTON_FOCUS_LABEL]>"
+                <md-button class="oppia-learner-continue-button protractor-test-continue-to-next-card-button" style="background-color: rgb(29, 59,164)" focus-on="<[CONTINUE_BUTTON_FOCUS_LABEL]>"
```
Ditto.

edallison replied:
Done

seanlip commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -53,8 +53,9 @@ md-input-group.md-default-theme label
-  color: #009688;
+.oppia-learner-continue-button {
+  color: #FFFFFF;
+  background-color: rgb(29, 59,164);
```
Nit: add single space after ,

edallison replied:
Done

seanlip commented:
Looks good! One issue though: *something* should happen to the button when the mouse hovers over it, or if it's in focus. The basic [.md-button class](https://material.angularjs.org/0.9.4/#/api/material.components.button/directive/mdButton) (from angular-material.css) does that already, so you might just need to change the colour for the corresponding `:hover` selector as well.

edallison commented:
I've addressed all of the issues however my latest commit isn't behaving properly with the new .md-button.oppia-learner-continue-button class. I've tried using hover, active, and focus selectors to get the button background to display correctly to no avail.  The only way I've been able to get the button background color to display correctly is by using !important on the background-color attribute which i know isn't any better than inline styling. Any suggestions?

Also for the hover and focus actions, would dimming the color of the button be alright?

seanlip commented:
Hi @edallison, note that immediately after the learner submits an answer, the continue button is focused, so that the learner can just click Enter to move to the next card. So if it looks like .md-button.oppia-learner-continue-button isn't working, it may actually be the :focus state that isn't working.

The way to investigate this is to use the browser console to investigate which CSS class's background-color is being applied. I did this and noticed that there's a `.md-button.md-default-theme:not([disabled]):hover` class that gets in the way. So you could try something more specific than that, like:

```
button.md-button.md-default-theme.oppia-learner-continue-button:hover,
button.md-button.md-default-theme.oppia-learner-continue-button:focus 
```

With regards to what the color of the Continue button should be on hover, I defer to @amitdeutsch. Amit, can you comment? (I'm a bit concerned that just making it lighter makes it look like the button is disabled.)

amitdeutsch commented:
I recommend making the button a slightly lighter blue when it's on hover or
focus:

#115FD4

On Thu, Sep 17, 2015 at 10:17 PM Sean Lip <notifications@github.com> wrote:

> Hi @edallison <https://github.com/edallison>, note that immediately after
> the learner submits an answer, the continue button is focused, so that the
> learner can just click Enter to move to the next card. So if it looks like
> .md-button.oppia-learner-continue-button isn't working, it may actually be
> the :focus state that isn't working.
>
> The way to investigate this is to use the browser console to investigate
> which CSS class's background-color is being applied. I did this and noticed
> that there's a .md-button.md-default-theme:not([disabled]):hover class
> that gets in the way. So you could try something more specific than that,
> like:
>
> button.md-button.md-default-theme.oppia-learner-continue-button:hover,
> button.md-button.md-default-theme.oppia-learner-continue-button:focus
>
> With regards to what the color of the Continue button should be on hover,
> I defer to @amitdeutsch <https://github.com/amitdeutsch>. Amit, can you
> comment? (I'm a bit concerned that just making it lighter makes it look
> like the button is disabled.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oppia/oppia/pull/973#issuecomment-141347423>.
>

edallison commented:
It should look and behave right now.

seanlip commented:
LGTM, thanks @edallison! Merging.
